samples/transactional-session/aspnetcore-webapi/SqlPersistenceTS_6/WebApplication/SendMessageController.cs

```diff
@@ -18,6 +21,7 @@ public SendMessageController(ITransactionalSession messageSession, MyDataContext
 
     #region txsession-controller
     [HttpGet]
+    [RequiresTransactionalSession]
     public async Task<string> Get()
```

Comment:
```suggestion
    public async Task<string> Get(ITransactionalSession messageSession)
```

How about we remove the constructor injection for `ITransactionalSession` (and the class filed) and add it explicitly as a parameter to the action?

This might be a better way to show that `ITransactionalSession` is usable only in actions with proper attribute applied.

Reply:
@tmasternak it could be. I left it as is because with the changes, I added in place the context will automatically be switched over to the non-managed connection and the transactional session is in an unusable state but since it is not used everything will just work. I felt like leaving it like that is probably more representative of how messy certain types of controllers are in the wild in terms of what is used where. Thoughts?

Reply:
> I felt like leaving it like that is probably more representative of how messy certain types of controllers are in the wild in terms of what is used where

Makes sense to me.

Reply:
@tmasternak I reflected about this again and went with your suggestion. Everything gets way cleaner and I think it is OK to be opinionated in the sample.